webui-src/app/people/people_ownids.js

```diff
@@ -1,213 +1,379 @@
+                else pseudonimous = true;
+              },
+            },
+            [
+              m("option[value=false]", "Linked to your Profile"),
+              m("option[value=true]", "Pseudonymous"),
+            ]
+          ),
+        ]
       ),
-      m('br'),
-      m('input[type=password][placeholder=Retroshare Passphrase]', {
+      m("br"),
+      m("input[type=password][placeholder=Retroshare Passphrase]", {
         value: pgpPassword,
-        oninput: e => (pgpPassword = e.target.value),
+        oninput: (e) => (pgpPassword = e.target.value),
       }),
       m(
-        'p',
-        'You can have one or more identities. ' +
-          'They are used when you chat in lobbies, ' +
-          'forums and channel comments. ' +
-          'They act as the destination for distant chat and ' +
-          'the Retroshare distant mail system.',
+        "p",
+        "You can have one or more identities. " +
+          "They are used when you chat in lobbies, " +
+          "forums and channel comments. " +
+          "They act as the destination for distant chat and " +
+          "the Retroshare distant mail system."
       ),
       m(
-        'button',
+        "button",
         {
-          onclick: () =>
-            rs.rsJsonApiRequest(
-              '/rsIdentity/createIdentity',
-              {
-                name,
-                pseudonimous,
-                pgpPassword,
-              },
-              data => {
-                const message = data.retval
-                  ? 'Successfully created identity.'
-                  : 'An error occured while creating identity.';
-                widget.popupMessage([
-                  m('h3', 'Create new Identity'),
-                  m('hr'),
-                  message,
-                ]);
-                m.redraw();
-              },
-            ),
+          onclick: () => {
+            if (pseudonimous == false) {
+              let paraphase = "";
+              widget.popupMessage([
+                m("i.fas.fa-user-edit"),
+                m("h3", "Enter your passpharse"),
+                m("hr"),
+
+                m("input[type=password][placeholder=Passpharse]", {
+                  style: "margin-top:50px;width:80%",
+                  oninput: (e) => {
+                    paraphase = e.target.value;
+                  },
+                }),
+                m(
+                  "button",
+                  {
+                    style: "margin-top:160px;",
+                    onclick: () => {
+                      rs.rsJsonApiRequest(
+                        "/rsIdentity/getOwnSignedIds",
+                        {},
+                        (owns) => {
+                          if (owns.ids.length > 0) {
+                            console.log(owns.ids);
+                            id = owns.ids[0];
+                            console.log(id);
+                          }
```

Comment:
What would happen if the list of ids is empty? It looks like you are using the `id` var without initializing it.

Reply:
@rottencandy, I have one doubt. Is it possible to have a retroshare account without any signed identity?.  we need to pass a retro share signed id for creating a signed identity. If we will not pass the id then retro share will not create a signed identity.

![createidentity](https://user-images.githubusercontent.com/64456168/113652419-6272ca80-96b1-11eb-88fa-7bb58558aecd.JPG)

Reply:
Not entirely sure but it seems possible to delete all identities associated with a profile. So this would be undefined behavior then?

Perhaps we should include the rsIdentity/createIdentiy call inside the condition check.

cc: @csoler

Reply:
Then , I will include rsIdentity/createIdentiy call inside the condition check.